crates/bifrost/src/providers/replicated_loglet/tasks/check_seal.rs

```diff
@@ -65,59 +84,146 @@ impl CheckSealTask {
-
-            let Some(next_node) = nodes.pop() else {
-                info!(
-                    loglet_id = %my_params.loglet_id,
-                    status = %nodeset_checker,
-                    effective_nodeset = %effective_nodeset,
-                    replication = %my_params.replication,
-                    "Insufficient nodes responded to GetLogletInfo requests, we cannot determine seal status, we'll assume it's unsealed for now",
-                );
-                return Ok(CheckSealOutcome::ProbablyOpen);
+            let request = GetLogletInfo {
+                header: LogServerRequestHeader::new(
+                    my_params.loglet_id,
+                    known_global_tail.latest_offset(),
+                ),
             };
 
-            let task = FindTailOnNode {
-                node_id: next_node,
-                loglet_id: my_params.loglet_id,
-                get_loglet_info_rpc,
-                known_global_tail,
-            };
-            let tail_status = tokio::select! {
-                _ = &mut cancel => {
-                    return Err(ShutdownError);
+            inflight_requests.spawn({
+                let networking = networking.clone();
+                let rpc_router = get_loglet_info_rpc.clone();
+                let known_global_tail = known_global_tail.clone();
+                let local_tail = local_tails.get(&node_id).cloned();
+
+                async move {
+                    let task = RunOnSingleNode::new(
+                        node_id,
+                        request,
+                        &rpc_router,
+                        &known_global_tail,
+                        // do not retry
+                        RetryPolicy::None,
+                    );
+
+                    (
+                        node_id,
+                        task.run(on_info_response(local_tail), &networking).await,
+                    )
                 }
-                (_, tail_status) = task.run(networking) => { tail_status },
+                .in_current_tc()
+                .in_current_span()
+            });
+        }
+
+        // Waiting for GetLogletInfo responses
+        loop {
+            // are we fully sealed? We use BestEffort here because we'd still want to consider a
+            // loglet sealed even if some nodes lost data. The check-seal wouldn't determine the
+            // known_global_tail, only whether the loglet is sealed or not and it's safe to do so.
+            // non-authoritative nodes cannot accept normal writes, only repair writes.
+            if nodeset_checker.check_fmajority(|attr| *attr) >= FMajorityResult::BestEffort {
+                // no need to detach, we don't need responses from the rest of the nodes
+                return Ok(CheckSealOutcome::FullySealed);
+            }
+
+            if nodeset_checker.check_fmajority(|attr| !(*attr)) >= FMajorityResult::BestEffort {
+                return Ok(CheckSealOutcome::Open);
+            }
```

Comment:
Why are we no longer treating a single sealed status as sufficient for returning `Sealing`?

Reply:
Is this to avoid that multiple clients start a sealing wave once they detect that a single node is sealed (even though the original sealer is still active and working on finishing the sealing operation)?

Reply:
If this is true, could the following situation now happen: The CC decided to seal a log (3 nodes, replication 2). It manages to seal N3 but then failed (let's assume the sequencer runs on N1). As long as all nodes are still reachable `CheckSealTask` would return `CheckSealOutcome::Open` and the sequencer will continue working as if nothing has happened just that it would no longer include `N3` in the writes. Is this a scenario where we would eventually like to finish the seal and reconfigure the log?

Reply:
In the case you described, the sequencer will still try and write to N3 (if it's alive) and if it observes that it's sealed, it'll mark the loglet as "maybe_sealed", this will force any future find_tail to runt he check seal. Now, whether checkSeal will not observe the seal to finish it off as you described.  In practice, the original requester of the seal operation will likely retry because the seal failed, but I can also remove this check from here to make this more pessimistic and let it wait for all responses before it responds. I'll go with the latter and evaluate based on how it performs.